src/context/SWR.tsx

```diff
@@ -0,0 +1,81 @@
+import { SWRConfig } from 'swr';
+import { BaseComponentProps } from 'types';
+
+const middleware = {
+    cache: () => {
+        // When initializing, we restore the data from `localStorage` into a map.
+        const loggingMap = {
+            data: new Map(
+                JSON.parse(localStorage.getItem('app-cache') ?? '[]')
+            ),
+            delete: (key: unknown) => {
+                console.log('SWR:cache:del', {
+                    key,
+                });
+
+                return loggingMap.data.delete(key);
+            },
+            get: (key: unknown) => {
+                const response = loggingMap.data.get(key);
+
+                console.log('SWR:cache:get', {
+                    key,
+                    response,
+                });
+
+                return response;
+            },
+            set: (key: unknown, val: unknown) => {
+                console.log('SWR:cache:set', {
+                    key,
+                    val,
+                });
+
+                return loggingMap.data.set(key, val);
+            },
+        };
+
+        // Before unloading the app, we write back all the data into `localStorage`.
+        window.addEventListener('beforeunload', () => {
+            const appCache = JSON.stringify(
+                Array.from(loggingMap.data.entries())
+            );
+            localStorage.setItem('app-cache', appCache);
+        });
```

Comment:
For local dev and testing having a local copy or stuff in local storage is nice. We need to look into this for prod and decide if we want to do this with the produt.